server/command.go

```diff
@@ -10,6 +10,13 @@ import (
 	"github.com/go-mysql-org/go-mysql/utils"
 )
 
+// StmtFieldsProvider is an optional interface that prepared statement contexts can implement
+// to provide field definitions for proxy passthrough scenarios.
+type StmtFieldsProvider interface {
```

Comment:
Just to verify why we need this interface? If the only reason is to cast `st.Context`, I think maybe we should move the common struct of client and server package to a common package, like utils or a new package, to avoid a runtime cast.

Reply:
Are you suggesting that we create something like

```
type StmtFields struct {
    ParamFields  []*Field
    ClumnFields []*Field
}
```

and then embed it or use it in `client.Stmt`?